### PR TITLE
Testing for dolt add --patch

### PR DESCRIPTION
**Pull request by macneale4**

### CODE REVIEW

**Comment by coffeegoddd** [comment]
@macneale4 __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>8c41e08</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>8c41e08</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Comment by coffeegoddd** [comment]
@macneale4 __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>955197c</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>955197c</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

File: integration-tests/bats/add-patch-expect/no_then_a.expect [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+source  "$env(BATS_CWD)/helper/common_expect_functions.tcl"
+
+spawn dolt add -p
+
+# This test will reject the first change for each of the 3 tables, then accept the rest.
+
+expect_with_defaults_2 {| \+ | 0  | Yellow | 255 | 255   | 0    |}    {Stage this row \[y,n,q,a,d,s,\?\]\? } { send "n\r"; }
+
+expect_with_defaults_2 {| - | 2  | Green | 0   | 255   | 0    |}      {Stage this row \[y,n,q,a,d,s,\?\]\? } { send "ar"; }
```
Comment (fulghum) [inline integration-tests/bats/add-patch-expect/no_then_a.expect:14]: Should this one be `send "a\r";`?

File: integration-tests/bats/add-patch-expect/restart_multiple_times.expect [hunk 2]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/expect
+
+set timeout 5
+set env(NO_COLOR) 1
+
+source  "$env(BATS_CWD)/helper/common_expect_functions.tcl"
+
+spawn dolt add --patch
+
+# This is a long script, but the idea is simple. Input y,n,y,s repeatedly and insure that the right prompts are seen
```
Comment (fulghum) [inline integration-tests/bats/add-patch-expect/restart_multiple_times.expect:10]: (minor) ;-) 

```suggestion
# This is a long script, but the idea is simple. Input y,n,y,s repeatedly and ensure that the right prompts are seen
```

**Review by fulghum — APPROVED** [review]
Looks good! 🚢

File: integration-tests/bats/add-patch-expect/no_then_a.expect (hunk 1 above)
Comment (macneale4) [reply integration-tests/bats/add-patch-expect/no_then_a.expect:14]: oh my. good catch. interesting that the test till passes!!!!

**Comment by coffeegoddd** [comment]
@macneale4 __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>6f7ab49</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>6f7ab49</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Pull request merged**